StackExchange.Redis/StackExchange/Redis/Position.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System;
+
+namespace StackExchange.Redis
+{
+    /// <summary>
+    /// A position within a stream. Defaults to <see cref="Position.New"/>.
+    /// </summary>
+    public struct Position
+    {
+        /// <summary>
+        /// Indicate a position from which to read a stream.
+        /// </summary>
+        /// <param name="readAfter">The position from which to read a stream.</param>
+        public Position(RedisValue readAfter)
+        {
+            if (readAfter == RedisValue.Null) throw new ArgumentNullException(nameof(readAfter), "readAfter cannot be RedisValue.Null.");
+
+            Kind = PositionKind.Explicit;
+            ExplicitValue = readAfter;
+        }
+
+        private Position(PositionKind kind)
+        {
+            Kind = kind;
+            ExplicitValue = RedisValue.Null;
+        }
+
+        private PositionKind? Kind { get; }
```

Comment:
In what scenario can this actually be null? The only way I can see is `default(Position)` - in that case, we should make the zero-value of `Kind` be whatever we want the default to be

Reply:
Yes, it would only be null with `default(Position)`, I suppose I was going out of my way to make it super-explicit with the nullable field. I will update the enum with the appropriate zero value, I had already rearranged the order of the enum values so the implicit zero value would be `PositionKind.Beginning` but I will make it explicit.